### PR TITLE
[23145] Restyle right elements in main header

### PR DESCRIPTION
**Pull request by HDinger**
This changes the top menu elements on the right side. The help icon gets a circle like on the website The user name is replaced by the users avatar. If there is no avatar the gravatar is shown. Similar to the home screen the user icon is shown if the plugin is missing.

https://community.openproject.com/work_packages/23145/activity

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hdinger-feature-23145-restyle-righ-elements-in-main-header.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://hdinger-feature-23145-restyle-righ-elements-in-main-header.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

**Comment by HDinger** [comment]
#### Before

bildschirmfoto 2016-05-20 um 11 00 40

#### After

bildschirmfoto 2016-05-20 um 12 26 28

File: spec/features/users/create_spec.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@
     end
   end
 
-  context 'with internal authentication' do
+  context 'with internal authentication', js: true do
```
Comment (ulferts) [inline spec/features/users/create_spec.rb]: is that `js: true` still needed or was it just added while fixing the spec?

File: spec/features/users/create_spec.rb [hunk 2]
```diff
@@ -126,7 +126,6 @@
           click_button 'Sign in'
 
           expect(page).to have_text 'My page'
-          expect(page).to have_text 'bobfirst boblast'
```
Comment (ulferts) [inline spec/features/users/create_spec.rb]: Please don't remove this line. You could replace it with a different selector e.g.

```
expect(page).to have_selector("[title='bobfirst boblast']")
```

Comment (ulferts) [inline spec/features/users/create_spec.rb]: Or you could use that 
`
have_link
`
method you employed elsewhere.

File: features/users/user_auth.feature [hunk 3]
```diff
@@ -36,7 +36,7 @@ Feature: User Authentication
   Scenario: A user is able to login successfully with provided credentials
     Given I am on the login page
     And I am admin
-    Then I should see "Admin" within "#top-menu-items"
+    Then I should find ".icon-user" within "#top-menu-items"
```
Comment (ulferts) [inline features/users/user_auth.feature]: This change leads to cukes that are way harder to read. Please write a small step definition, e.g.

```
Then I should see "Admin" as being logged in
```

Let me know if you need pointers as to how it is done.

File: features/users/status.feature [hunk 4]
```diff
@@ -66,22 +66,25 @@ Feature: User Status
     And I click "Unlock"
     Then I should not see "bobby"
     And I try to log in with user "bobby"
-    Then I should see "Bob Bobbit"
+    Then I should find ".icon-user" within "#account-nav-right"
 
+  @javascript
```
Comment (ulferts) [inline features/users/status.feature]: Are those selenium markers (`@javascript`) needed?

**Comment by ulferts** [comment]
Change itself is looking fine. 

The failing tests are not caused by the PR. They have been fixed in dev in the meantime.

Once the minor remarks are handled, this can be merged in.

**Comment by HDinger** [comment]
@ulferts I included the things you mentioned. Could you have a look at the new step? It seems to work right, but I'm not quite sure with the implementation.

**Comment by oliverguenther** [comment]
@HDinger  Head's up: I've rebased on dev and force pushed this.

Feature-wise, this looks good to merge now. The icon may have some contrast issues depending on what your actual avatar is. I've resolved some code styling issues. Merging after travis agrees.

_No avatar, no gravatar_
bildschirmfoto 2016-06-13 um 14 28 06

_Gravatar default_
bildschirmfoto 2016-06-13 um 14 28 11

_My avatar_
bildschirmfoto 2016-06-13 um 14 28 12

**Pull request merged**